src/dataStructures/linkedList/LinkedList.java

```diff
@@ -214,29 +215,36 @@ public Node<T> get(int idx) {
   }
 
   /**
-   * reverse the linked list
+   * reverse the linked list.
+   * A good video to visualise this algorithm can be found
+   * <a = https://www.youtube.com/watch?v=D7y_hoT_YZI, href = "url">here</a>.
    */
   public void reverse() {
-    if (head == null || head.next == null) {
+    if (head == null || head.next == null) { // No need to reverse if list is empty or only 1 element.
       return;
     }
 
     Node<T> prev = head;
     Node<T> curr = head.next;
-    Node<T> newHead = curr;
-    prev.next = null;
+    Node<T> newHead = curr; // Store the next head.
+    prev.next = null; // Reset to null, representing end of list.
     while (curr.next != null) {
-      newHead = curr.next;
-      curr.next = prev;
+      newHead = curr.next; // We set the next element as the newHead.
+      curr.next = prev; // Replace our current node as the previous node.
       prev = curr;
-      curr = newHead;
+      curr = newHead; // Set our current node as the next element (newHead) to look at.
     }
     newHead.next = prev;
-    head = newHead;
+    head = newHead; // newHead is last ele from org. list -> First ele of our reversed list.
   }
 
   /**
-   * sorts the linked list by their natural order
+   * sorts the linked list by their natural order.
+   * Generally, merge sort is the most efficient sorting algorithm for linked lists.
+   * This is because of the nature of merge sort, which requires lesser lookups.
```

Comment:
Great that you tackled this point on why merge sort in selected. While merge sort in general (average-case) does require less comparisons than other comparison-based sorting algor, the time complexity remains at O(nlogn). 

The more pressing issue here is due to accessing a random node in the linked list being a costly operation (recall this is O(n) in the worst case as you mentioned in the README). This makes other algorithms like quicksort and heapsort  (its ok if u dk how they work but basically they rely on selecting an element that could be anywhere in the array, or linked list) inefficient since the array version would have look-up in O(1).

Merge sort resolves this because the merge routine simply compares the elements at the front which remains easily accessed via a linked list.